mimic/test/test_nova.py

```diff
@@ -39,6 +39,45 @@ def setUp(self):
         self.server_id = self.create_server_response_body['server']['id']
         self.nth_endpoint_public = helper.nth_endpoint_public
 
+    def test_create_server_with_manual_diskConfig(self):
+        """
+        Servers should respect the provided OS-DCF:diskConfig setting if
+        supplied.
+        """
+        create_server = request(
+            self, self.root, "POST", self.uri + '/servers',
+            json.dumps({
+                "server": {
+                    "name": self.server_name + "A",
+                    "imageRef": "test-image",
+                    "flavorRef": "test-flavor",
+                    "OS-DCF:diskConfig": "MANUAL",
+                }
+            }))
+        create_server_response = self.successResultOf(create_server)
+        response_body = self.successResultOf(
+            treq.json_content(create_server_response))
+        self.assertEqual(
+            response_body['server']['OS-DCF:diskConfig'], 'MANUAL')
+
+    def test_create_server_with_bad_diskConfig(self):
+        """
+        Given a diskConfig not in ["AUTO", "MANUAL"], we expect a 400
+        Bad Request error.
+        """
```

Comment:
The test docstring idiom I tend to prefer would phrase this as an action verb where the system under test is the subject, such as:

```python
"""
When ``create_server`` is passed an invalid ``OS-DCF:diskConfig`` that is not `"AUTO"` or `"MANUAL"`,
it returns an HTTP status code of 400 (bad request).
"""
```